File: backend/entity/Education.go
package entity

import "gorm.io/gorm"

type Education struct {
	gorm.Model
	
	Year             int       `json:"year"`
	Grade            float64   `json:"grade"`
	
	UniversityID     uint       `json:"university_id"`
	University       University `gorm:"foreignKey:UniversityID"`

	FacultyID        uint     `json:"faculty_id"`
	Faculty          Faculty  `gorm:"foreignKey:FacultyID"`

	ProgramID        uint     `json:"program_id"`
	Program          Program  `gorm:"foreignKey:ProgramID"`

	StudentID        uint     `json:"student_id"`
	Student          Student  `gorm:"foreignKey:StudentID"`

	EducationLevelID uint           `json:"education_level_id"`
	EducationLevel   EducationLevel `gorm:"foreignKey:EducationLevelID"`
}

File: frontend/src/interfaces/Education.ts
import type { UniversityInterface } from "./UniversityInterface";
import type { FacultyInterface } from "./FacultyInterface";
import type { ProgramInterface } from "./ProgramInterface";
import type { EducationLevelInterface } from "./EducationLevel";

export interface EducationInterface {
  id?: number;
  user_id: number;

  university_id?: number;
  University: UniversityInterface;

  faculty_id?: number;
  Faculty: FacultyInterface;

  program_id?: number;
  Program: ProgramInterface;

  year: number;

  education_level_id?: number; // เพิ่ม ID สำหรับ join
  EducationLevel: EducationLevelInterface; // แทน string เดิม

  grade: number;
}
